src/App.js

```diff
@@ -4,31 +4,56 @@ import { getAll, getById } from './api/phone'
 import Basket from './Basket'
 import Filter from './Filter'
 import Catalog from './Catalog'
+import Viewer from './Viewer'
 
 import './App.css';
 
-
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       phones: getAll(),
       selectedPhone: null,
-      basketItems: [],
+      basketItems: {
+       
+      },
     };
+    
+  }
+
+  handleRemoveFromBasket = (phoneName) => {
+    this.setState((prev) => {
+      const copyBasketItems = {...prev.basketItems};
+      const count = copyBasketItems[phoneName] || 0;
+      count > 1 ? copyBasketItems[phoneName] = count -1: delete copyBasketItems[phoneName]
+
+      return { basketItems: copyBasketItems}
+    }) 
   }
 
+  handleAddToBasket = (phoneName) => { 
+    this.setState((prev)=> {
+      const quantity = prev.basketItems[phoneName] || 0;
+      const copyBasketItems = {...prev.basketItems};
+      copyBasketItems[phoneName] = quantity + 1;
+      return { basketItems: copyBasketItems}
```

Comment:
```suggestion
      return { basketItems: {
        ...prev.basketItems,
        [phoneName]: count + 1,
      }}
```
This way you can get rid of lines 38 & 39